packages/multichain-account-service/src/providers/SnapAccountProvider.ts

```diff
@@ -92,6 +96,103 @@ export abstract class SnapAccountProvider extends BaseBip44AccountProvider {
       });
   }
 
+  #getKeyringClientFromSnapId(snapId: string): KeyringClient {
+    return new KeyringClient({
+      send: async (request: JsonRpcRequest) => {
+        const response = await this.messenger.call(
+          'SnapController:handleRequest',
+          {
+            snapId: snapId as SnapId,
+            origin: 'metamask',
+            handler: HandlerType.OnKeyringRequest,
+            request,
+          },
+        );
+        return response as Json;
+      },
+    });
+  }
+
+  async resyncAccounts(
+    accounts: Bip44Account<InternalAccount>[],
+  ): Promise<void> {
+    const localSnapAccounts = accounts.filter(
+      (account) =>
+        account.metadata.snap && account.metadata.snap.id === this.snapId,
+    );
+    const snapAccounts = new Set(
+      (await this.client.listAccounts()).map((account) => account.id),
+    );
+
+    // We want this part to be fast, so we only check for sizes, but we might need
+    // to make "diff" between the 2 states to not miss any de-sync.
+    if (localSnapAccounts.length > snapAccounts.size) {
```

Comment:
It might be worth logging if `localSnapAccounts.length < snapAccounts.size`, which also shouldn't happen.